### PR TITLE
Add AddParticipantsViewController which support adding team members to conversations

### PR DESCRIPTION
**Pull request by typfel**
https://wearezeta.atlassian.net/browse/ZIOS-8583

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    
+    let searchIcon = UIImageView()
+    let titleLabel : UILabel = UILabel()
+    let tokenField : TokenField = TokenField()
+    let closeButton : IconButton
+    let userSelection : UserSelection
+    let colorSchemeVariant : ColorSchemeVariant
+    
+    @objc
+    public var delegate : SearchHeaderViewControllerDelegate? = nil
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift]: Should it be `weak` ?

Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift]: Should not cause memory leak right now, but I guess having it weak is more robust.

File: Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        
+        super.init(nibName: nil, bundle: nil)
+        
+        userSelection.add(observer: self)
+    }
+    
+    public override func viewDidLoad() {
+        searchIcon.image = UIImage(for: .search, iconSize: .tiny, color: UIColor.wr_color(fromColorScheme: ColorSchemeColorTokenFieldTextPlaceHolder, variant: colorSchemeVariant))
+        
+        titleLabel.text = title
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift]: Seems unnecessary, looking on the next line.

File: Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    
+    fileprivate func createConstraints() {
+        
+        constrain(view, titleLabel, closeButton, searchIcon, tokenField) { view, titleLabel, closeButton, searchIcon, tokenField in
+            titleLabel.top == view.top + 28
+            titleLabel.leading == tokenField.leading
+            titleLabel.trailing == tokenField.trailing
+            
+            tokenField.top == titleLabel.bottom + 16
+            tokenField.left == view.left + 8
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift]: Should it be `leading` / `trailing`?

File: Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    public func resetQuery() {
+        tokenField.filterUnwantedAttachments()
+        delegate?.searchHeaderViewController(self, updatedSearchQuery: tokenField.filterText)
+    }
+    
+}
+
+extension SearchHeaderViewController : UserSelectionObserver {
+    
+    public func userSelection(_ userSelection: UserSelection, wasReplacedBy users: [ZMUser]) {
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift]: Should this case be handled too?

File: Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift (hunk 2 above)
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift]: :)

File: Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift (hunk 3 above)
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift]: Don't know how the TokenField behaves, just copied this from the ShareViewController

File: Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift (hunk 4 above)
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift]: No, that would case a loop. Can add a comment.

File: Wire-iOS/Sources/UserInterface/StartUI/StartUI/StartUIViewController.m [hunk 5]
```diff
@@ -412,26 +400,31 @@ - (void)searchResultsController:(SearchResultsController *)searchResultsControll
...
-#pragma mark - PeopleInputControllerDelegate
+#pragma mark - SearchHeaderViewControllerDelegate
 
-- (void)peopleInputController:(PeopleInputController *)controller changedFilterTextTo:(NSString *)text
+- (void)searchHeaderViewControllerDidCancelAction:(SearchHeaderViewController *)searchHeaderViewController
 {
-    [self.searchResultsController cancelPreviousSearch];
-    [NSObject cancelPreviousPerformRequestsWithTarget:self selector:@selector(performSearch) object:nil];
-    [self performSelector:@selector(performSearch) withObject:nil afterDelay:0.3f];
+    [self.searchHeaderViewController resignFirstResponder]; // FIXME jacob
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/StartUI/StartUI/StartUIViewController.m]: If TODO is not needed I guess it can be removed.

**Pull request merged**